islandora-indexing-fcrepo/src/main/java/ca/islandora/alpaca/indexing/fcrepo/FcrepoIndexer.java

```diff
@@ -18,89 +18,50 @@
 import static org.apache.camel.LoggingLevel.ERROR;
 import static org.apache.camel.LoggingLevel.INFO;
-import static org.apache.camel.LoggingLevel.DEBUG;
+import static org.apache.camel.LoggingLevel.TRACE;
+import static org.apache.camel.LoggingLevel.WARN;
 import static org.slf4j.LoggerFactory.getLogger;
 
-import ca.islandora.alpaca.support.event.AS2Event;
 import org.apache.camel.Exchange;
 import org.apache.camel.Predicate;
-import org.apache.camel.PropertyInject;
-import org.apache.camel.builder.RouteBuilder;
 import org.apache.camel.builder.PredicateBuilder;
-import org.apache.camel.http.common.HttpOperationFailedException;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.http.base.HttpOperationFailedException;
 import org.apache.camel.model.dataformat.JsonLibrary;
 import org.slf4j.Logger;
-// import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import ca.islandora.alpaca.support.event.AS2Event;
+import ca.islandora.alpaca.support.exceptions.MissingCanonicalUrlException;
+import ca.islandora.alpaca.support.exceptions.MissingJsonUrlException;
+import ca.islandora.alpaca.support.exceptions.MissingJsonldUrlException;
 
 /**
+ * Camel Route to index Drupal nodes into Fedora.
+ *
  * @author Danny Lamb
+ * @author whikloj
  */
-// @JsonIgnoreProperties(ignoreUnknown = true)
 public class FcrepoIndexer extends RouteBuilder {
 
-    /**
-     * Header to use to pass the Fedora Base URI on.
-     */
-    public final static String FEDORA_HEADER = "X-Islandora-Fedora-Endpoint";
-
-    /**
-     * Maximum attempts to deliver a message.
-     */
-    @PropertyInject("error.maxRedeliveries")
-    private int maxRedeliveries;
-
-    /**
-     * Base URI of the milliner web service.
-     */
-    @PropertyInject("milliner.baseUrl")
-    private String millinerBaseUrl;
+    @Autowired
+    private FcrepoIndexerOptions config;
 
     /**
      * The Logger.
      */
     private static final Logger LOGGER = getLogger(FcrepoIndexer.class);
 
-    /**
-     * @return  Number of times to retry
-     */
-    public int getMaxRedeliveries() {
-        return maxRedeliveries;
-    }
-
-    /**
-     * @param   maxRedeliveries Number of times to retry
-     */
-    public void setMaxRedeliveries(final int maxRedeliveries) {
-        this.maxRedeliveries = maxRedeliveries;
-    }
-
-    /**
-     * @return  Milliner base url
-     */
-    public String getMillinerBaseUrl() {
-        return enforceTrailingSlash(millinerBaseUrl);
-    }
-
-    /**
-     * @param   millinerBaseUrl Milliner base url
-     */
-    public void setMillinerBaseUrl(final String millinerBaseUrl) {
-        this.millinerBaseUrl = millinerBaseUrl;
-    }
-
-    private String enforceTrailingSlash(final String baseUrl) {
-        final String trimmed = baseUrl.trim();
-        return trimmed.endsWith("/") ? trimmed : trimmed + "/";
-    }
-
-
     @Override
     public void configure() {
```

Comment:
PMD's complaining about this now:

```
/home/adam/repos/Alpaca/islandora-indexing-fcrepo/src/main/java/ca/islandora/alpaca/indexing/fcrepo/FcrepoIndexer.java 	59 	Avoid really long methods.
```

Linking to https://pmd.github.io/pmd-6.21.0/pmd_rules_java_design.html#excessivemethodlength

... Not sure if it's the additions here pushing it over whatever threshold, or if it would've reported previously.

Reply:
Yeah that is a good question, the only way I can think of would be to move some of the processing into separate `Processor` classes. There isn't much in the way of re-usable code, but some of the property setting might work. Unsure how important everyone thinks this rule is

Reply:
Could... possibly just suppress the warning more specifically? https://pmd.github.io/latest/pmd_userdocs_suppressing_warnings.html

... as in, looks like this might do  the trick?
```suggestion
    @SuppressWarnings("PMD.ExcessiveMethodLength")
    public void configure() {
```

... including an inline comment with a bit of rationale might make sense, but yeah, I don't see any great issue going that way, if we want.... I don't see breaking up into a number of different private/protected methods which setup the different bits as being excessively useful, over the one bigger method.